hasher-matcher-actioner/hmalib/models.py

```diff
@@ -281,3 +283,56 @@ def from_time_range(
             & Key("UpdatedAt").between(start_time, end_time),
             ProjectionExpression=cls.DEFAULT_PROJ_EXP,
         ).get("Items", [])
+    @staticmethod
+    def from_dynamodb_string(string: str) -> "Label":
+        return Label(*string.split(":"))
+
+
+@dataclass
+class MatchMessage(SNSMessage):
+    """
+    Captures a match that will need to be processed. Joins the dataset and the
+    pipeline together.
+
+    Pipeline fields:
+    - `content_key`: A way for partners to refer uniquely to content on their
+      site
+    - `content_hash`: The pipeline generated hash for the content_key
```

Comment:
Wait, if we are sending this to the pipeline how did it get a pipeline hash?

Reply:
We're not sending this to the pipeline. One phase (matcher) of the pipeline generates messages for the subsequent phase (actions). But they all need some 'pipeline-unique-ids' to make sense of the information. These fields provide those.

Reply:
Is "matched hash"/"matched_signal" a synonym for "pipeline hash"?

What's the difference between content id/key and pipeline id?